python/rikai/types/geometry.py

```diff
@@ -440,3 +448,207 @@ def to_dict(self) -> dict:
             "height": self.height,
             "heading": self.heading,
         }
+
+
+class Mask(ToNumpy, ToDict):
+    """2-d Mask over an image
+
+    This 2D mask can be built from:
+
+    - A binary-valued (0 or 1) 2D-numpy matrix (:py:class:`Mask.Type.MASK`).
+    - A Run Length Encoded (RLE) data. It supports both row-based RLE
+      (:py:class:`Mask.Type.RLE`) or column-based RLE
+      (:py:class:`Mask.Type.COCO_RLE`) which is used in the Coco dataset.
+    - A Polygon ``[x0, y0, x1, y1, ..., xn, yn]``
+
+    Parameters
+    ----------
+    data: list or :py:class:`np.ndarray`
+        The mask data. Can be a numpy array or a list.
+    height: int, optional
+        The height of the image this mask applies to.
+    width: int, optional
+        The width of the image this mask applies to.
+    mask_type: :py:class:`Mask.Type`
+        The type of the mask.
+
+    Examples
+    --------
+
+    .. code-block:: python
+
+        from pycocotools.coco import COCO
+        from rikai.types import Mask
+
+        coco = COCO("instance_train2017.json")
+        ann = coco.loadAnns(ann_id)
+        image = coco.loadImgs(ann["image_id"])
+        if ann["iscrowed"] == 0:
+            mask = Mask.from_polygon(
+                ann["segmentation"],
+                height=image["height"],
+                width=image["width],
+            )
+        else:
+            mask = Mask.from_coco_rle(
+                ann["segmentation"]["counts"],
+                height=image["height"],
+                width=image["width],
+            )
+
+    """
+
+    __UDT__ = MaskType()
+
+    class Type(Enum):
+        """Mask type."""
+
+        MASK = 0
```

Comment:
this is a bit confusing that you can have a Mask.Type be MASK

Reply:
What about naming it to `Mask.Type.RAW`? or sth else?
So basically, polygon and RLE are compressed mask, and this `Mask.Type.MASK` is the original 2d matrix.

Reply:
yeah, I understand the intended behavior, it's just the naming.
Calling it "RAW" implies it's lower-level than the other types. Off the top of my head I can suggestion something like calling the class `Segmentation` and keep the `MASK` constant name. OR keep the `Mask` class name and change the constant to `Mask.Type.DENSE` or something?

Is there community convention on this already that makes sense? If there is we can just follow it?

Reply:
This is up to discuss. So a "Segmentation" is similar to a "Detection" IMO, where it represents the output of a "segmentation task".  I am not certain to name it Mask.TYPE.MASK or .DENSE is better.

Let me run some reference with other ppl then.

Reply:
Talked offline, lets just not support the raw / dense mask type for now.  RLE is loss less and statistically much smaller than the mask anyway. Lets always store data in RLE and polygon forms as the same as coco.